### PR TITLE
use graphql for generating upload sessions

### PR DESCRIPTION
**Pull request by dsokal**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary.

# Why

https://github.com/expo/universe/pull/7279 adds GraphQL mutation for generating upload sessions.

# How

I replaced the REST API call with the GraphQL mutation call.

# Test Plan

I ran `eas build` and I noticed the project archive got uploaded to S3.

### CODE REVIEW

File: packages/eas-cli/src/uploads.ts [hunk 1]
```diff
@@ -33,25 +28,8 @@ export async function uploadAsync(
...
-  const { data } = await apiClient
-    .post('upload-sessions', {
-      json: {
-        type: uploadType,
-        checksum: fileHash,
-      },
-    })
-    .json();
-  return data.presignedUrl;
+async function obtainS3PresignedPostAsync(type: UploadSessionType): Promise<PresignedPost> {
```
Comment (wkozyra95) [inline packages/eas-cli/src/uploads.ts:31]: maybe remove that function?

File: packages/eas-cli/src/uploads.ts [hunk 2]
```diff
@@ -2,15 +2,10 @@ import assert from 'assert';
 import FormData from 'form-data';
 import fs from 'fs';
 import got, { Progress } from 'got';
-import md5File from 'md5-file';
```
Comment (wkozyra95) [inline packages/eas-cli/src/uploads.ts:5]: this package is not used anywhere else, you can remove it from package.json

File: packages/eas-cli/src/uploads.ts [hunk 3]
```diff
@@ -2,15 +2,10 @@ import assert from 'assert';
 import FormData from 'form-data';
 import fs from 'fs';
```
Comment (wkozyra95) [inline packages/eas-cli/src/uploads.ts:3]: unrelated to this PR, but maybe switch to fs-extra

**Review by wkozyra95 — APPROVED** [review]
remember to fix tests

File: packages/eas-cli/src/uploads.ts (hunk 2 above)
Comment (dsokal) [reply packages/eas-cli/src/uploads.ts:5]: Done

**Pull request merged**